xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/api/Document.java

```diff
@@ -3158,4 +3158,14 @@ public boolean isTranslation()
     {
         return 1 == this.getDoc().getTranslation();
     }
+
+    /**
+     * @return the maximum authorized length for a document full name (see {@link #getFullName()}).
+     * @since 11.3RC1
+     */
+    @Unstable
+    public int getNameMaxLength()
+    {
+        return this.doc.getFullNameMaxLength();
```

Comment:
This is a weird API. I would never have expected to see this in the Document or XWikiDocument classes....

It seems to be it's related to the store.

Reply:
> At an doc API level I'm not expecting to have any constraint on the doc full name.

I'm not sure to understand your POV there: there's a technical constraint on the name size which was until now hidden, but it has always been there. Here I'm just exposing it to the user so they can use it to create some specific checks.